components/renku_data_services/session/blueprints.py

```diff
@@ -26,9 +34,7 @@ def get_all(self) -> BlueprintFactoryResponse:
 
         async def _get_all(_: Request) -> JSONResponse:
             environments = await self.session_repo.get_environments()
-            return json(
-                [apispec.Environment.model_validate(e).model_dump(exclude_none=True, mode="json") for e in environments]
-            )
+            return validated_json(apispec.EnvironmentList, [self._dump_environment(e) for e in environments])
```

Comment:
question: does `validate_and_dump` not work for a list of models? we use it that way in other places in the code and it seems to work fine. or why is `_dump_environment` needed? e.g. `GroupsBP.get_all`

Reply:
I've solved that with a custom serializers in the past: https://github.com/SwissDataScienceCenter/renku-data-services/blob/main/components/renku_data_services/namespace/apispec_base.py#L20 though if it can be `None` is also need a check for that to not end up with `'None'` as an id

Reply:
Oh this would explain why it works on the `id` field but not with other fields. I don't know if we can really use this approach since it does not look like it can be easily be applied only to the classes which need it.

Reply:
yes it applies to everything and not per class, unfortunately. You could use `isinstance()` to only have it apply to ULID and pass everything else on verbatim

Reply:
See c2525fdca46b856a1f1b8336e1f95bbe3b5eb584.